include/zephyr/drivers/spi.h

```diff
@@ -765,6 +888,7 @@ static inline bool spi_is_ready_dt(const struct spi_dt_spec *spec)
  *
  * @retval frames Positive number of frames received in slave mode.
  * @retval 0 If successful in master mode.
+ * @retval -ENOTSUP or -EINVAL means some part of the spi config is not possible for device
```

Comment:
If we want to be consistent here, -ENOTSUP is the only one to fit this description. -EINVAL means that something's off  (I don't know like requesting peripheral mode with loopback for instance?)

Apply that where relevant

Reply:
If a driver is using both in a wrong way, it will be a driver problem. Doc should not reflect the misuse of error codes in drivers, we'll fix the drivers instead.

Reply:
ok, I split these into two separate meanings on next push that I think more accurately reflects both the intended meaning of the error codes and how drivers are actually using them